configserver/src/main/java/com/yahoo/vespa/config/server/session/SessionRepository.java

```diff
@@ -572,8 +572,16 @@ private void childEvent(CuratorFramework ignored, PathChildrenCacheEvent event)
     public void deleteExpiredSessions(Map<ApplicationId, Long> activeSessions) {
         log.log(Level.FINE, () -> "Purging old sessions for tenant '" + tenantName + "'");
         Set<LocalSession> toDelete = new HashSet<>();
+        Set<Long> newSessions = findNewSessionsInFileSystem();
         try {
             for (LocalSession candidate : getLocalSessionsFromFileSystem()) {
+                // Skip sessions newly added (we might have a session in the file system, but not in ZooKeeper,
+                // we don't want to touch any of them)
+                if (newSessions.contains(candidate.getSessionId())) {
+                    log.log(Level.INFO, () -> "Skipping session " + candidate.getSessionId() + ", newly created: ");
```

Comment:
Nit: Since INFO is always logged, there is no need to pass a String supplier.

Reply:
Originally had it at level FINE, so forgot when changing. I'll change it to FINE in a few days if this change works as expected